server/api/utils/background_tasks/internal.py

```diff
@@ -230,3 +172,32 @@ def _generate_background_task(
         return mlrun.common.schemas.BackgroundTask(
             metadata=metadata, spec=spec, status=status
         )
+
+    def _get_active_task_name_by_kind(self, kind: str):
+        return self._background_tasks_kind_locks.get(kind, (None, None))[0]
+
+    def _get_previous_task_name_by_kind(self, kind: str):
```

Comment:
When/where is this being used? 
and why are we saving the previous task at all?

Reply:
You're right it currently isn't anything using this method. Though once I implement delete project again, I see a use case for a scenario where you delete project X. And after it is deleted, you might want to inspect the job that deleted it. That endpoint will use this method (Once the task is no longer active, it is immediately previous).